netbox/ipam/forms/model_forms.py

```diff
@@ -818,13 +814,28 @@ def __init__(self, *args, **kwargs):
         if (parent_object_type_id := get_field_value(self, 'parent_object_type')):
             try:
                 parent_type = ContentType.objects.get(pk=parent_object_type_id)
+                if parent_type.model_class() == Device:
+                    self.fields['ipaddresses'].widget.add_query_params({
+                        'device_id': '$parent',
+                    })
+                elif parent_type.model_class() == VirtualMachine:
+                    self.fields['ipaddresses'].widget.add_query_params({
+                        'virtual_machine_id': '$parent',
+                    })
+                elif parent_type.model_class() == FHRPGroup:
+                    self.fields['ipaddresses'].widget.add_query_params({
+                        'fhrpgroup_id': '$parent',
+                    })
                 model = parent_type.model_class()
```

Comment:
You can move this up and reference `model` when determining the query parameters to attach.

Reply:
Missed that one